adal/src/main/java/com/microsoft/aad/adal/SSOStateSerializer.java

```diff
@@ -62,15 +62,18 @@
             .registerTypeAdapter(TokenCacheItem.class, new TokenCacheItemSerializationAdapater())
             .create();
 
-    private int getVersion() {
-        return version;
+    /**
+     * No args constructor for use in serialization for Gson to prevent usage of sun.misc.Unsafe
+     */
+    @SuppressWarnings("unused")
```

Comment:
@weijjia @heidijinxujia I already explained this here: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/842#issuecomment-305906759, just two comments above. 

The unit tests were failing because of using the default/package private constructor. Making them private failed the tests since they could not be initialized.